source/core_http_client.c

```diff
@@ -701,19 +700,12 @@ static int httpParserOnHeaderValueCallback( http_parser * pHttpParser,
                                             size_t length )
 {
     HTTPParsingContext_t * pParsingContext = NULL;
-    HTTPResponse_t * pResponse = NULL;
-
-    /* Disable unused variable warning. */
-    ( void ) pResponse;
 
     assert( pHttpParser != NULL );
     assert( pHttpParser->data != NULL );
     assert( pLoc != NULL );
 
     pParsingContext = ( HTTPParsingContext_t * ) ( pHttpParser->data );
-    pResponse = pParsingContext->pResponse;
-
-    assert( pResponse != NULL );
```

Comment:
Why wasn't the assert retained as
`assert( pParsingContext->pResponse != NULL );`?